README

#=========================================
DIANA
Version: 3.16.4
(Using metlibs Version 2009-12-02)
#=========================================

Norwegian Meteorological Institute (met.no)
Box 43 Blindern
0313 OSLO
NORWAY

email: diana@met.no

##################################################

Contents:
1. About Diana
2. Install
3. Running Diana
4. Configuration
5. File formats

#-------------------------------------------
# 1. About Diana
#-------------------------------------------

Diana is the core visualisation tool from met.no (Norwegian
Meteorological Institute) and is distributed under the GPL
license. See gpl.txt for details concerning the GPL license.

Diana is a graphical viewer and editor developed for use with
meteorological and oceanographic data.  It uses OpenGL for the
graphics and Trolltech Qt for the user interface.

Diana shows fields, observations and satellite and radar images on a
map. 2D-trajectories can be computed from wind and ocean current
fields.  The tool also displays vertical profiles (soundings),
vertical crossections and wave spectrum data in separate windows, all
from preprocessed data. Preprocessing software is however not a part
of Diana.  The editor tools consist of a field editor and a drawing
tool for fronts, symbols and area types.

Diana supports multiple languages via the Qt linguist
system. Additional languages can be added by preparing new
diana/lang/diana_XX.ts files (using the Qt linguist program) and
running "make languages"

In addition to the interactive tool, there is also a command line
version of Diana for batch plotting (bdiana).


++++++++++++++


#-------------------------------------------
# 2. Install
#-------------------------------------------

see INSTALL

Demo data
=================================================================
#Download demodata from 
ftp://ftp.met.no/diana/diana_demodata.tar.gz
#Download maps from 
ftp://ftp.met.no/diana/diana_maps.tar.gz
# unpack data files
tar xzvf diana_demodata.tar.gz
tar xzvf diana_maps.tar.gz


#-------------------------------------------
# 3 Running Diana
#-------------------------------------------


1. Prepare setupfile with exact installation directory:
Edit DIANADIR, OPDATA, MAPDIR

2. Starting Diana:
 diana.bin            # looks for diana.setup in current directory
 # or
 bin/diana -s /etc/diana/diana.setup

 Language is preset to English (Norwegian is the only other option, in
 the setup file).  User guide available under Help.

-----------------

Running bdiana:
run "bin/bdiana -example" and study the Diana User Guide bdiana chapter.

#-------------------------------------------
# 4 Configuration files for diana
#-------------------------------------------

All specifications of data access (incl. background map files) is in
etc/diana/diana.setup

Changing the setup file:
Example:To add a new field file to diana.setup, 
put the following inside the <FIELD_FILES>
 section:
 #----------------------------------------------
 filegroup=SURGE
 m=TOPO f=/disk1/SURGE/ETOPO2/bathymetry.flt  t=MetnoFieldFile
 #-----------------------------------------------

 If you need to add a new field parameter, the
 <METNOFIELDFILE_PARAMETERS> section is the correct place . You will also
 need to add a PLOT comand for your new parameter, see the
 <FIELD_PLOT> section.


#-------------------------------------------
# 5 File formats for diana
#-------------------------------------------

The current version supports following file formats

Fields:                MetnoFieldFile (proprietary met.no)
                       NetCDF
                       Grib
Observations:          MetnoObs       (proprietary met.no)
	               BUFR
                       ascii
Image (sat.,radar):    mitiff         (proprietary met.no)
                       HDF5
Prognostic sounding:   vprof          (proprietary met.no)
vertical crossections: vcross         (proprietary met.no)
Wave spectrum:         wamspec        (proprietary met.no)
Maps:                  shape
		       ascii
		       metno-format   (documentation available)














debian/changelog
diana (3.28.0-ubuntu1) lucid; urgency=low

  * using fimex 0.37.1
  * using metlibs_trunk
  * compiled without profet support
	
 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 29 Mar 2012 21:42:51 +0200

diana (3.27.4-ubuntu1) lucid; urgency=low

  * Observations - bugfix criterias

 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 01 Mar 2012 18:55:51 +0100

diana (3.27.3-ubuntu1) lucid; urgency=low

  * HQC - bugfix

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 27 Feb 2012 14:31:58 +0100

diana (3.27.2-ubuntu1) lucid; urgency=low

  * New EPS parameter - cluster representative
  * Bugfix profet: computed fields (like FF) were not
	changed when the original fields (u,v) were edited. 

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 14 Feb 2012 20:32:42 +0100

diana (3.27.1-ubuntu1) lucid; urgency=low

  * Minor fixes

 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 09 Feb 2012 14:48:14 +0100

diana (3.27.0-ubuntu1) lucid; urgency=low

  * Using metlibs_4.3
  * Using libfimex-0.35
  * Using WDB/fimex
  * Reading from KDVH

 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 26 Jan 2012 09:42:51 +0100

diana (3.26.4-ubuntu3) lucid; urgency=low

  * bdiana: default canvas changed to QGLPixelBuffer

 -- Lisbeth Bergholt <lisbethb@pc2982.pc.met.no>  Mon, 28 Nov 2011 09:42:43 +0100

diana (3.26.4-ubuntu2) lucid; urgency=low

  * using +towgs in all proj strings except old mitiff files.
  * Using metlibs_4.2
  * Using libfimex-0.33-1

 -- Lisbeth Bergholt <lisbethb@vpn-116-99.met.no>  Sat, 26 Nov 2011 16:42:14 +0100

diana (3.26.4-ubuntu1) lucid; urgency=low

  * Bugfix:DifferenceField - plot string was not parsed correctly
  * Using metlibs_trunk (by mistake)

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 22 Nov 2011 13:54:05 +0100

diana (3.26.3-ubuntu1) lucid; urgency=low

  * Bugfix: Ascii-observations - only observations from one file were used
  * Using metlibs_trunk (by mistake)

 -- Lisbeth Bergholt <lisbethb@.met.no>  Mon, 21 Nov 2011 13:30:46 +0100

diana (3.26.2-ubuntu1) lucid; urgency=low

  * Bugfix: mark measurements/trajectory positions

 -- Lisbeth Bergholt <lisbethb@met.no>  Sun, 09 Oct 2011 17:28:50 +0200

diana (3.26.1-ubuntu2) lucid; urgency=low

  * Bugfix bdiana: vertical levels

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 27 Sep 2011 14:09:07 +0200

diana (3.26.1-ubuntu1) lucid; urgency=low

  * Bugfix bdiana: translation from plot name to field name

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 27 Sep 2011 12:52:03 +0200

diana (3.26.0-ubuntu2) lucid; urgency=low

  * Projections: datum fix removed

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 26 Sep 2011 10:39:04 +0200

diana (3.26.0-ubuntu1) lucid; urgency=low

  * Using metlibs_4.2
  * Using libfimex-0.31-2
  * New profet function
  * Improved fimex reading
  * Vertical crossections: vt+omega / vt+w fixed
  * Bufr: skip time variables with missing valaues 
  * Bugfix:projections

 -- Lisbeth Bergholt <lisbethb@met.no>  Fri, 23 Sep 2011 15:38:14 +0200

diana (3.25.5-ubuntu2) lucid; urgency=low

  * Bugfix: bdiana did not find fieldTimes

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 04 Jul 2011 12:41:27 +0200

diana (3.25.5-ubuntu1) lucid; urgency=low

  * Bugfix: Labels from ObsAscii files
  * Using metlibs_4.1
  * Using libfimex-dev 0.29

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 27 Jun 2011 09:45:07 +0200

diana (3.25.4-ubuntu2) lucid; urgency=low

  * Bugfix was not included in last version

 -- Lisbeth Bergholt <lisbethb@met.no>  Wed, 22 Jun 2011 08:52:03 +0200

diana (3.25.4-ubuntu1) lucid; urgency=low

  * Bugfix: Memory leak in StationPlot
  * PostScript fix: Always set font-specification

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 21 Jun 2011 14:53:45 +0200

diana (3.25.3-ubuntu1) lucid; urgency=low

  * More bugfix: profet

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 20 Jun 2011 13:30:54 +0200

diana (3.25.2-ubuntu2) lucid; urgency=low

  * Bugfix case insensitivity

 -- Lisbeth Bergholt <lisbethb@met.no>  Fri, 17 Jun 2011 14:54:36 +0200

diana (3.25.2-ubuntu1) lucid; urgency=low

  * Bugfix xylimit
  * Reverted fix: model and field are now case insensitive

 -- Lisbeth Bergholt <lisbethb@met.no>  Fri, 17 Jun 2011 08:28:03 +0200

diana (3.25.1-ubuntu1) lucid; urgency=low

  * Field: model and field are now case sensitive

 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 16 Jun 2011 11:26:45 +0200

diana (3.25.0-ubuntu4) lucid; urgency=low

  * Using metlibs_4.1
  * Using libfimex-dev 0.28-2

 -- Lisbeth Bergholt <lisbethb@met.no>  Wed, 15 Jun 2011 10:06:40 +0200

diana (3.25.0-ubuntu3) lucid; urgency=low

  * Using metlibs_4.1
  * Using libfimex-dev 0.28

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 10 May 2011 08:31:50 +0200

diana (3.25.0-ubuntu2) lucid; urgency=low

  * Geotiff support enabled
  * Using metlibs_4.1

 -- Lisbeth Bergholt <lisbethb@met.no>  Thu, 05 May 2011 12:02:30 +0200

diana (3.25.0-ubuntu1) lucid; urgency=low

  * New field reader using fimex
  * Using metlibs_2011-01-28

 -- Lisbeth Bergholt <lisbethb@met.no>  Fri, 28 Jan 2011 12:47:24 +0100

diana (3.24.0-ubuntu2) lucid; urgency=low

  * Bugfix: enable/disable ObjectPlots
  * Using metlibs_2011-01-13

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 24 Jan 2011 11:01:52 +0100

diana (3.24.0-ubuntu1) lucid; urgency=low

  * Using metlibs_2011-01-13
  * new version of profet

 -- Lisbeth Bergholt <lisbethb@met.no>  Fri, 21 Jan 2011 13:26:08 +0100

diana (3.23.3-ubuntu1) lucid; urgency=low

  * Bugfix parsing logfile
  * Using metlibs_2010-11-15

 -- Lisbeth Bergholt <lisbethb@vpn-117-63.met.no>  Mon, 06 Dec 2010 09:00:58 +0100

diana (3.23.2-ubuntu1) lucid; urgency=low

  * Bugfix obsplot: pressure tendency, temperature precision, metar clouds 
  * Bugfix editing: Snow_showers -> Snow showers

 -- Lisbeth Bergholt <lisbethb@met.no>  Mon, 29 Nov 2010 13:22:29 +0100

diana (3.23.1-ubuntu6) lucid; urgency=low

  * Added export LANG=en_US in diana.in

 -- Lisbeth Bergholt <lisbethb@met.no>  Tue, 23 Nov 2010 14:00:45 +0100


dianaPROFF.sh
#!/bin/sh
set -x

input=$0
dianadir="/metno/local/dianaPROFF"
test -d $dianadir || dianadir="/usr/local/diana"

export LD_LIBRARY_PATH=/metno/local/lib/mesa:$LD_LIBRARY_PATH
## export LD_LIBRARY_PATH=$dianadir/lib:$LD_LIBRARY_PATH

export DIANADIR=$dianadir

OPSYS=`uname -s`
case $OPSYS in
    IRIX*) metsys="SGI"
	   debugger="dbx"
	   SUBNET=`/usr/etc/ifconfig ec0 | grep "inet" | cut -d. -f3`
	   ;;
    *)	   metsys="Linux"
	   test $dianadir = "/usr/local/diana" && metsys="Linux-Debian"
	   debugger="gdb"
	   SUBNET=`/sbin/ifconfig eth0 | grep "inet addr" | cut -d. -f3`
	   ulimit -c 100000000000
	   ;;
esac

case $SUBNET in
    20)    region="FOU" ;;
    36|40) region="VV" ;;
    48)    region="VNN" ;;
### 50)                        region="MA"  ;;
    16|18|24|26|56|90|104|50) region="VA"  ;;
    *)                         region="VTK" ;;
esac

home=$HOME/grun_op
test -d $home || mkdir $home

home=$home/diana
test -d $home           || mkdir $home
test -d $home/bin       || mkdir $home/bin
test -d $home/bin/work  || mkdir $home/bin/work

cd $home/bin

rm core* 1>/dev/null 2>&1
test -s dbxresult  && mv -f dbxresult dbxresult.old
test -s mail.core  && mv -f mail.core mail.core.old

# remove old lpr print files, left after failure...
find . -name "prt_????-??-??_??:??:??.ps" -mtime +1 -exec rm {} \;

#--------------------------------------------
# to be removed:
ln -sf $dianadir/etc/ANAborders.* .
ln -sf $dianadir/etc/synpltab.dat .
ln -sf $dianadir/etc/metpltab.dat .
#--------------------------------------------
#ln -sf $dianadir/etc/profet.setup-${region} diana.setup
ln -sf $dianadir/etc/diana.setup-${region} diana.setup

tstart=`date`

$dianadir/bin/diana -p -style cleanlooks 
#-s $dianadir/etc/profet.setup-${region}

tstop=`date`

corefound="no"
for corefile in core*
do
    test -f $corefile || continue
    test -s $corefile || continue
    corefound="yes"
done

if [ $corefound = "yes" ] ; then

echo "USER $USER"     >mail.core
echo "-------------" >>mail.core
uname -a             >>mail.core
echo "-------------" >>mail.core
echo "START $tstart" >>mail.core
echo "STOP  $tstop"  >>mail.core
echo "DianaPROFF version (diana.news):" >>mail.core 
cat diana.news       >>mail.core
echo "-------------" >>mail.core
pwd                  >>mail.core
ls -ltr core*        >>mail.core

for corefile in core*
do

test -f $corefile || continue
test -s $corefile || continue

$debugger $dianadir/bin/diana $corefile 1>dbxresult 2>&1 <<ENDDBX
where 
y
y
y
n
q
ENDDBX

echo "===============================================" >>mail.core
cat dbxresult        >>mail.core

rm $corefile

done

echo "===============================================" >>mail.core
hinv                 >>mail.core

adr="audun.christoffersen@met.no helen.korsmo@met.no lisbeth.bergholt@met.no"
Mail -s "DIANA CRASH" $adr < mail.core

fi

exit
